frontend/packages/console-plugin-sdk/src/api/useDynamicPluginInfo.ts

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { useForceRender } from '@console/shared/src/hooks/useForceRender';
+import { DynamicPluginInfo } from '../store';
+import { subscribeToDynamicPlugins } from './pluginSubscriptionService';
+
+/**
+ * React hook for consuming Console dynamic plugin runtime information.
+ *
+ * When the runtime status of a dynamic plugin changes, the React component
+ * is re-rendered with the hook returning an up-to-date plugin information.
+ *
+ * The hook's result is guaranteed to be referentially stable across re-renders.
+ *
+ * @returns Console dynamic plugin runtime information.
+ */
+export const useDynamicPluginInfo = (): DynamicPluginInfo[] => {
+  const forceRender = useForceRender();
+
+  const isMountedRef = React.useRef(true);
+  const unsubscribeRef = React.useRef<VoidFunction>(null);
+  const pluginInfoEntriesRef = React.useRef<DynamicPluginInfo[]>([]);
```

Comment:
Couldn't we use `useState` and avoid the need for `useForceRender` ?
Looks like this is a similar pattern to `useExtenions`.

Reply:
this was only a suggestion but it seems based on the requirement to compute an initial state through other means, a ref is needed.